src/api.py

```diff
@@ -1,15 +1,42 @@
+from functools import wraps
+
 from sanic import Blueprint, response
 from sanic.response import json
-from sanic_jwt.decorators import inject_user, protected
 from sanic_jwt.exceptions import AuthenticationFailed
 
 from src.utils import expects_json_object
 
 blueprint = Blueprint("root", version="v1")
 
 
-def auth_required(func):
-    return (inject_user(blueprint))(protected(blueprint)(func))
+def auth_required(f):
+    @wraps(f)
+    async def decorated_function(request, *args, **kwargs):
+        def get_token(header):
```

Comment:
I would prefer manipulating the `request.headers['Authorization']` directly instead of creating another fn here